lang/py/src/avro/schema.py

```diff
@@ -722,14 +818,27 @@ def make_avsc_object(json_data, names=None):
   if hasattr(json_data, 'get') and callable(json_data.get):
     type = json_data.get('type')
     other_props = get_other_props(json_data, SCHEMA_RESERVED_PROPS)
+    logical_type = None
+    if 'logicalType' in json_data:
+      logical_type = json_data.get('logicalType')
+    if logical_type and logical_type != 'decimal':
+      raise SchemaParseException("Currently does not support %s logical type" % logical_type)
+    else:
+      precision = json_data.get('precision')
+      scale = 0 if json_data.get('scale') is None else json_data.get('scale')
     if type in PRIMITIVE_TYPES:
+      if type == 'bytes' and logical_type == 'decimal':
+        return DecimalSchema(type, precision, scale)
```

Comment:
I'm ok with this approach for now. This is probably not going to be easy to scale once more logical types are introduced though (v1.8.0 introduces more date/time family logical types). Something worth to think ahead.